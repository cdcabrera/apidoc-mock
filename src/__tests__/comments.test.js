const comments = require('../comments');

//   "generateApiData": [Function],
//   "getCommentContentStatusType": [Function],
//   "parseCommentTag": [Function],
//   "readFilesGetComments": [Function],

describe('comments', () => {
  it('should return specific properties', () => {
    expect(comments).toMatchSnapshot('specific properties');
  });

  it('should filter for ignored headers', () => {
    expect({
      default: comments.filterForIgnoredHeaders(),
      basic: comments.filterForIgnoredHeaders({ 'access-control-lorem': 'ipsum', dolor: 'sit' })
    }).toMatchSnapshot('filtered headers');
  });

  it('should parse comment header examples', () => {
    const mockFunction = jest.fn();
    const mockLogger = {
      warn: mockFunction
    };

    expect({
      default: comments.parseCommentHeaderExample(),
      basic: comments.parseCommentHeaderExample({
        content: '{ "Access-control-lorem": "ipsum", "dolor": "" }',
        type: 'request'
      }),
      malformedStringJSON: comments.parseCommentHeaderExample(
        {
          content: '{ dolor: "sit" }'
        },
        { logger: mockLogger }
      ),
      malformedObjectJSON: comments.parseCommentHeaderExample(
        {
          content: { dolor: 'sit' }
        },
        { logger: mockLogger }
      )
    }).toMatchSnapshot('parsed headers');

    expect(mockFunction.mock.calls.map(mock => mock.map(call => call.split(/},\s/)[0]))).toMatchSnapshot(
      'malformed string JSON'
    );
  });

  it('should parse comment content, status, and type', () => {
    const mockFunction = jest.fn();
    const mockLogger = {
      warn: mockFunction
    };

    expect({
      default: comments.parseCommentContentStatusType(),
      basic: comments.parseCommentContentStatusType({
        content: `HTTP/1.1 400\n{ "bad": "request example" }`
      }),
      fallbackStatus: comments.parseCommentContentStatusType({
        content: `HTTP/1.1\n{ "lorem": "ipsum" }`,
        status: 201
      }),
      withErrorStatusAndSuccessExample: comments.parseCommentContentStatusType(
        {
          content: `HTTP/1.1 400\n{ "lorem": "ipsum", "dolor": "sit", "hello": "world" }`,
          isSuccessExample: true
        },
        {
          logger: mockLogger
        }
      ),
      withSuccessStatusAndErrorExample: comments.parseCommentContentStatusType(
        {
          content: `HTTP/1.1 200\n{ "lorem": "ipsum", "dolor": "sit", "hello": "world" }`,
          isSuccessExample: false
        },
        {
          logger: mockLogger
        }
      ),
      existingContentType: comments.parseCommentContentStatusType({
        content: `HTTP/1.1 200\n{ lorem: "ipsum"}`,
        type: 'application/lorem+ipsum'
      })
    }).toMatchSnapshot('parsed content');

    expect(mockFunction.mock.calls).toMatchSnapshot('mismatched status codes');

    expect(
      ['zip', 'gzip', 'json', 'xml', 'html', 'csv', 'css', 'svg', 'txt', 'lorem'].map(type => ({
        type,
        output: comments.parseCommentContentStatusType({
          type
        })
      }))
    ).toMatchSnapshot('content types');
  });

  it('should normalize response behavior with mock settings', () => {
    expect({
      default: comments.normalizeMockSettings(),
      delayNumber: comments.normalizeMockSettings({ delay: 500 }),
      forceNumber: comments.normalizeMockSettings({ Force: 418 }),
      response: comments.normalizeMockSettings({ response: 'lorem ipsum' }),
      random: comments.normalizeMockSettings({ random: 'dolor sit' }),
      randomSuccess: comments.normalizeMockSettings({ RANDOMSUCCESS: 'dolor sit' }),
      randomError: comments.normalizeMockSettings({ randomError: 'dolor sit' })
    }).toMatchSnapshot('normalized');
  });

  it('should parse a comment tag', () => {
    expect({
      default: comments.parseCommentTag()
    }).toMatchSnapshot('empty');

    expect({
      default: comments.parseCommentTag({ tag: 'api', name: '/lorem/ipsum' }),
      basic: comments.parseCommentTag({ tag: 'api', name: '/lorem/ipsum', type: 'POST' })
    }).toMatchSnapshot('api');

    expect({
      default: comments.parseCommentTag({ tag: 'apiMock' }),
      basic: comments.parseCommentTag({ tag: 'apiMock', type: 'FORCE', name: 404 }),
      withName: comments.parseCommentTag({ tag: 'apiMock', type: 'FORCE', name: 'lorem-ipsum' })
    }).toMatchSnapshot('apiMock');

    expect({
      default: comments.parseCommentTag({ tag: 'apiHeader' }),
      basic: comments.parseCommentTag({ tag: 'apiHeader', name: 'Lorem-Ipsum' }),
      withType: comments.parseCommentTag({ tag: 'apiHeader', name: 'Lorem-Ipsum', type: 'request' }),
      withDescription: comments.parseCommentTag({ tag: 'apiHeader', name: 'Lorem-Ipsum', description: 'Dolor sit' })
    }).toMatchSnapshot('apiHeader');

    expect({
      default: comments.parseCommentTag({ tag: 'apiHeaderExample' }),
      basic: comments.parseCommentTag({
        tag: 'apiHeaderExample',
        name: 'Lorem-Ipsum',
        type: 'REQUEST',
        description: 'Dolor sit',
        source: [{ tokens: { description: '{ "DolorSit": "mock header value" }' } }]
      }),
      withMalformedJSON: comments.parseCommentTag({
        tag: 'apiHeaderExample',
        name: 'Lorem-Ipsum',
        type: 'RESPONSE',
        description: 'Dolor sit',
        source: [{ tokens: { description: '{ DolorSit: "mock header value", "lorem-ipsum": "mock header value" }' } }]
      }),
      withMissingSource: comments.parseCommentTag({
        tag: 'apiHeaderExample',
        name: 'Lorem-Ipsum',
        type: 'RESPONSE',
        description: 'Dolor sit',
        source: [{ tokens: undefined }]
      })
    }).toMatchSnapshot('apiHeaderExample');

    expect({
      default: comments.parseCommentTag({ tag: 'apiSuccessExample' }),
      basic: comments.parseCommentTag({
        tag: 'apiSuccessExample',
        type: 'json',
        name: '200',
        source: [{ tokens: { description: '{ "DolorSit": "mock response value" }' } }]
      }),
      withMissingStatus: comments.parseCommentTag({
        tag: 'apiSuccessExample',
        type: 'json',
        source: [{ tokens: { description: '{ "DolorSit": "mock response value" }' } }]
      }),
      withMalformedSource: comments.parseCommentTag({
        tag: 'apiSuccessExample',
        type: 'json',
        name: '200',
        source: [{ tokens: {} }]
      })
    }).toMatchSnapshot('apiSuccessExample');

    expect({
      default: comments.parseCommentTag({ tag: 'apiErrorExample' }),
      basic: comments.parseCommentTag({
        tag: 'apiErrorExample',
        type: 'json',
        name: '400',
        source: [{ tokens: { description: '{ "DolorSit": "mock response value" }' } }]
      }),
      withMissingStatus: comments.parseCommentTag({
        tag: 'apiErrorExample',
        type: 'json',
        source: [{ tokens: { description: '{ "DolorSit": "mock response value" }' } }]
      })
    }).toMatchSnapshot('apiErrorExample');
  });

  it('should attempt to generate a basic api spec', () => {
    const mockFunction = jest.fn();
    const mockLogger = {
      warn: mockFunction
    };

    expect({
      default: comments.generateApiSpec(),
      basic: comments.generateApiSpec([
        {
          tags: [
            [
              {
                tag: 'api',
                type: 'post',
                name: '/dolor/sit'
              },
              {
                tag: 'apiMock',
                type: 'force',
                name: '500'
              },
              {
                tag: 'apiHeader',
                name: 'Authorization',
                description: 'Dolor sit token'
              },
              {
                tag: 'apiHeaderExample',
                type: 'REQUEST',
                source: [{ tokens: { description: '{ "Access-control-mock": "lorem-ipsum", "Hello": "world" }' } }]
              },
              {
                tag: 'apiHeaderExample',
                type: 'response',
                source: [{ tokens: { description: '{ "Mock-response-header": "lorem-ipsum" }' } }]
              },
              {
                tag: 'apiHeaderExample',
                source: [{ tokens: {} }]
              },
              {
                tag: 'apiSuccessExample',
                type: 'json',
                name: '200',
                source: [{ tokens: { description: '{ "DolorSit": "mock response value" }' } }]
              },
              {
                tag: 'apiErrorExample',
                type: 'json',
                name: '400',
                source: [{ tokens: { description: '{ "DolorSit": "mock response value" }' } }]
              }
            ]
          ],
          file: './lorem-ipsum.txt'
        }
      ]),
      missingPath: comments.generateApiSpec(
        [
          {
            tags: [
              [
                {
                  tag: 'api',
                  type: 'post'
                }
              ]
            ],
            file: './lorem-ipsum.txt'
          }
        ],
        { logger: mockLogger }
      )
    }).toMatchSnapshot('generated spec');

    expect(mockFunction.mock.calls).toMatchSnapshot('missing api path');
  });

  it('should attempt to read files and get comments', () => {
    /*
    const mockFunction = jest.fn();
    const mockLogger = {
      error: mockFunction,
      warn: mockFunction
    };
    */

    const basicFixture = generateFixture(
      `/**
         * @api {get} /hello/world/
         * @apiSuccessExample {json} Success-Response:
         *     HTTP/1.1 200 OK
         *     {
         *       "success": "test"
         *     }
         * @apiErrorExample {json} Error-Response:
         *     HTTP/1.1 400 OK
         *     {
         *       "error": "test"
         *     }
         */`,
      { dir: './.fixtures/comments', filename: 'test.txt' }
    );

    expect({
      default: comments.readFilesGetComments(),
      basic: comments.readFilesGetComments([basicFixture.path])
    }).toMatchSnapshot('read files');

    // expect(mockFunction.mock.calls).toMatchSnapshot('logger');
  });

  it('should attempt to generate an api', () => {
    expect({
      default: comments.getDocs(),
      filesNoLength: comments.getDocs({ files: [] })
    }).toMatchSnapshot('no files');

    const mockFunction = jest.fn();
    comments.getDocs(
      { files: ['lorem/ipsum.txt'] },
      { generateApiSpec: mockFunction, readFilesGetComments: mockFunction }
    );
    expect(mockFunction).toHaveBeenCalledTimes(2);
  });
});
