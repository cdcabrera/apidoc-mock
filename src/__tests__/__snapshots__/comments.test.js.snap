// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments should attempt to generate a basic api spec: generated spec 1`] = `
{
  "basic": [
    {
      "errorExamples": [
        {
          "content": "{ "DolorSit": "mock response value" }",
          "status": 400,
          "type": "application/json",
        },
      ],
      "mockSettings": {
        "forceStatus": 500,
      },
      "requestHeaders": {
        "Authorization": "Dolor sit token",
        "Hello": "world",
      },
      "responseHeaders": {
        "Mock-response-header": "lorem-ipsum",
      },
      "successExamples": [
        {
          "content": "{ "DolorSit": "mock response value" }",
          "status": 200,
          "type": "application/json",
        },
      ],
      "type": "post",
      "url": "/dolor/sit",
    },
  ],
  "default": [],
  "missingPath": [],
}
`;

exports[`comments should attempt to generate a basic api spec: missing api path 1`] = `
[
  [
    "file-parse	:missing mock path after "@api" comment in ./lorem-ipsum.txt",
  ],
]
`;

exports[`comments should attempt to generate an api: no files 1`] = `
{
  "default": [],
  "filesNoLength": [],
}
`;

exports[`comments should attempt to read files and get comments: read files 1`] = `
{
  "basic": [
    {
      "file": ".fixtures/comments/test.txt",
      "tags": [
        [
          {
            "description": "",
            "name": "/hello/world/",
            "optional": false,
            "problems": [],
            "source": [
              {
                "number": 1,
                "source": "         * @api {get} /hello/world/",
                "tokens": {
                  "delimiter": "*",
                  "description": "",
                  "end": "",
                  "lineEnd": "",
                  "name": "/hello/world/",
                  "postDelimiter": " ",
                  "postName": "",
                  "postTag": " ",
                  "postType": " ",
                  "start": "         ",
                  "tag": "@api",
                  "type": "{get}",
                },
              },
            ],
            "tag": "api",
            "type": "get",
          },
          {
            "description": "HTTP/1.1 200 OK { "success": "test" }",
            "name": "Success-Response:",
            "optional": false,
            "problems": [],
            "source": [
              {
                "number": 2,
                "source": "         * @apiSuccessExample {json} Success-Response:",
                "tokens": {
                  "delimiter": "*",
                  "description": "",
                  "end": "",
                  "lineEnd": "",
                  "name": "Success-Response:",
                  "postDelimiter": " ",
                  "postName": "",
                  "postTag": " ",
                  "postType": " ",
                  "start": "         ",
                  "tag": "@apiSuccessExample",
                  "type": "{json}",
                },
              },
              {
                "number": 3,
                "source": "         *     HTTP/1.1 200 OK",
                "tokens": {
                  "delimiter": "*",
                  "description": "HTTP/1.1 200 OK",
                  "end": "",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "     ",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
              {
                "number": 4,
                "source": "         *     {",
                "tokens": {
                  "delimiter": "*",
                  "description": "{",
                  "end": "",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "     ",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
              {
                "number": 5,
                "source": "         *       "success": "test"",
                "tokens": {
                  "delimiter": "*",
                  "description": ""success": "test"",
                  "end": "",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "       ",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
              {
                "number": 6,
                "source": "         *     }",
                "tokens": {
                  "delimiter": "*",
                  "description": "}",
                  "end": "",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "     ",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
            ],
            "tag": "apiSuccessExample",
            "type": "json",
          },
          {
            "description": "HTTP/1.1 400 OK { "error": "test" }",
            "name": "Error-Response:",
            "optional": false,
            "problems": [],
            "source": [
              {
                "number": 7,
                "source": "         * @apiErrorExample {json} Error-Response:",
                "tokens": {
                  "delimiter": "*",
                  "description": "",
                  "end": "",
                  "lineEnd": "",
                  "name": "Error-Response:",
                  "postDelimiter": " ",
                  "postName": "",
                  "postTag": " ",
                  "postType": " ",
                  "start": "         ",
                  "tag": "@apiErrorExample",
                  "type": "{json}",
                },
              },
              {
                "number": 8,
                "source": "         *     HTTP/1.1 400 OK",
                "tokens": {
                  "delimiter": "*",
                  "description": "HTTP/1.1 400 OK",
                  "end": "",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "     ",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
              {
                "number": 9,
                "source": "         *     {",
                "tokens": {
                  "delimiter": "*",
                  "description": "{",
                  "end": "",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "     ",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
              {
                "number": 10,
                "source": "         *       "error": "test"",
                "tokens": {
                  "delimiter": "*",
                  "description": ""error": "test"",
                  "end": "",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "       ",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
              {
                "number": 11,
                "source": "         *     }",
                "tokens": {
                  "delimiter": "*",
                  "description": "}",
                  "end": "",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "     ",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
              {
                "number": 12,
                "source": "         */",
                "tokens": {
                  "delimiter": "",
                  "description": "",
                  "end": "*/",
                  "lineEnd": "",
                  "name": "",
                  "postDelimiter": "",
                  "postName": "",
                  "postTag": "",
                  "postType": "",
                  "start": "         ",
                  "tag": "",
                  "type": "",
                },
              },
            ],
            "tag": "apiErrorExample",
            "type": "json",
          },
        ],
      ],
    },
  ],
  "default": [],
}
`;

exports[`comments should filter for ignored headers: filtered headers 1`] = `
{
  "basic": {
    "dolor": "sit",
  },
  "default": undefined,
}
`;

exports[`comments should normalize response behavior with mock settings: normalized 1`] = `
{
  "default": {},
  "delayNumber": {
    "delay": 500,
  },
  "forceNumber": {
    "forceStatus": 418,
  },
  "random": {
    "reload": true,
    "response": "response",
  },
  "randomError": {
    "reload": true,
    "response": "error",
  },
  "randomSuccess": {
    "reload": true,
    "response": "success",
  },
  "response": {
    "response": "response",
  },
}
`;

exports[`comments should parse a comment tag: api 1`] = `
{
  "basic": {
    "type": "post",
    "url": "/lorem/ipsum",
  },
  "default": {
    "type": "get",
    "url": "/lorem/ipsum",
  },
}
`;

exports[`comments should parse a comment tag: apiErrorExample 1`] = `
{
  "basic": {
    "errorExample": {
      "content": "{ "DolorSit": "mock response value" }",
      "isSuccessExample": false,
      "status": 400,
      "type": "json",
    },
  },
  "default": {
    "errorExample": {
      "content": undefined,
      "isSuccessExample": false,
      "status": undefined,
      "type": "txt",
    },
  },
  "withMissingStatus": {
    "errorExample": {
      "content": "{ "DolorSit": "mock response value" }",
      "isSuccessExample": false,
      "status": undefined,
      "type": "json",
    },
  },
}
`;

exports[`comments should parse a comment tag: apiHeader 1`] = `
{
  "basic": {
    "headerExample": {
      "content": {
        "Lorem-Ipsum": "",
      },
      "type": undefined,
    },
  },
  "default": {},
  "withDescription": {
    "headerExample": {
      "content": {
        "Lorem-Ipsum": "Dolor sit",
      },
      "type": undefined,
    },
  },
  "withType": {
    "headerExample": {
      "content": {
        "Lorem-Ipsum": "",
      },
      "type": "request",
    },
  },
}
`;

exports[`comments should parse a comment tag: apiHeaderExample 1`] = `
{
  "basic": {
    "headerExample": {
      "content": "{ "DolorSit": "mock header value" }",
      "type": "request",
    },
  },
  "default": {
    "headerExample": {
      "content": undefined,
      "type": undefined,
    },
  },
  "withMalformedJSON": {
    "headerExample": {
      "content": "{ DolorSit: "mock header value", "lorem-ipsum": "mock header value" }",
      "type": "response",
    },
  },
  "withMissingSource": {
    "headerExample": {
      "content": "",
      "type": "response",
    },
  },
}
`;

exports[`comments should parse a comment tag: apiMock 1`] = `
{
  "basic": {
    "mock": {
      "force": 404,
    },
  },
  "default": {},
  "withName": {
    "mock": {
      "force": "lorem-ipsum",
    },
  },
}
`;

exports[`comments should parse a comment tag: apiSuccessExample 1`] = `
{
  "basic": {
    "successExample": {
      "content": "{ "DolorSit": "mock response value" }",
      "isSuccessExample": true,
      "status": 200,
      "type": "json",
    },
  },
  "default": {
    "successExample": {
      "content": undefined,
      "isSuccessExample": true,
      "status": undefined,
      "type": "txt",
    },
  },
  "withMalformedSource": {
    "successExample": {
      "content": "",
      "isSuccessExample": true,
      "status": 200,
      "type": "json",
    },
  },
  "withMissingStatus": {
    "successExample": {
      "content": "{ "DolorSit": "mock response value" }",
      "isSuccessExample": true,
      "status": undefined,
      "type": "json",
    },
  },
}
`;

exports[`comments should parse a comment tag: empty 1`] = `
{
  "default": {},
}
`;

exports[`comments should parse comment content, status, and type: content types 1`] = `
[
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "application/zip",
    },
    "type": "zip",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "application/gzip",
    },
    "type": "gzip",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "application/json",
    },
    "type": "json",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "text/xml",
    },
    "type": "xml",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "text/html",
    },
    "type": "html",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "text/csv",
    },
    "type": "csv",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "text/css",
    },
    "type": "css",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "image/svg+xml",
    },
    "type": "svg",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "text/plain",
    },
    "type": "txt",
  },
  {
    "output": {
      "content": "",
      "status": 200,
      "type": "text/plain",
    },
    "type": "lorem",
  },
]
`;

exports[`comments should parse comment content, status, and type: mismatched status codes 1`] = `
[
  [
    "file-parse	:400 :success :mismatched status and example, { "lorem": "ipsum",...",
  ],
  [
    "file-parse	:200 :error :mismatched status and example, { "lorem": "ipsum",...",
  ],
]
`;

exports[`comments should parse comment content, status, and type: parsed content 1`] = `
{
  "basic": {
    "content": "{ "bad": "request example" }",
    "status": 400,
    "type": "text/plain",
  },
  "default": {
    "content": "",
    "status": 200,
    "type": "text/plain",
  },
  "existingContentType": {
    "content": "{ lorem: "ipsum"}",
    "status": 200,
    "type": "application/lorem+ipsum",
  },
  "fallbackStatus": {
    "content": "{ "lorem": "ipsum" }",
    "status": 201,
    "type": "text/plain",
  },
  "withErrorStatusAndSuccessExample": {
    "content": "{ "lorem": "ipsum", "dolor": "sit", "hello": "world" }",
    "status": 400,
    "type": "text/plain",
  },
  "withSuccessStatusAndErrorExample": {
    "content": "{ "lorem": "ipsum", "dolor": "sit", "hello": "world" }",
    "status": 200,
    "type": "text/plain",
  },
}
`;

exports[`comments should parse comment header examples: malformed string JSON 1`] = `
[
  [
    "file-parse	:JSON :ignoring malformed "@apiHeaderExample" { dolor: "sit" }, Expected property name or '}' in JSON at position 2",
  ],
]
`;

exports[`comments should parse comment header examples: parsed headers 1`] = `
{
  "basic": {
    "content": {
      "dolor": "",
    },
    "type": "request",
  },
  "default": {
    "content": undefined,
    "type": undefined,
  },
  "malformedObjectJSON": {
    "content": {
      "dolor": "sit",
    },
    "type": undefined,
  },
  "malformedStringJSON": {
    "content": undefined,
    "type": undefined,
  },
}
`;

exports[`comments should return specific properties: specific properties 1`] = `
{
  "filterForIgnoredHeaders": [Function],
  "generateApiSpec": [Function],
  "getDocs": [Function],
  "normalizeMockSettings": [Function],
  "parseCommentContentStatusType": [Function],
  "parseCommentHeaderExample": [Function],
  "parseCommentTag": [Function],
  "readFilesGetComments": [Function],
}
`;
